backend/src/reactions/reactions.service.spec.ts

```diff
@@ -0,0 +1,253 @@
+    diariesService = module.get<DiariesService>(DiariesService);
+  });
+
+  describe('getAllReaction', () => {
+    beforeEach(() => jest.clearAllMocks());
+
+    it('diary id에 대한 리액션 및 리액션의 사용자 정보 조회', async () => {
+      const diaryId = 1;
+      const reactions = [
+        {
+          userId: 1,
+          nickname: 'test1',
+          profileImage: 'test1',
+          reaction: '💜',
+        },
+        {
+          userId: 2,
+          nickname: 'test1',
+          profileImage: 'test1',
+          reaction: '🤩',
+        },
+      ];
+
+      (reactionsRepository.findByDiary as jest.Mock).mockResolvedValue(reactions);
+
+      expect(await reactionsService.getAllReaction(diaryId)).toBe(reactions);
+      expect(reactionsRepository.findByDiary).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('saveReaction', () => {
+    beforeEach(() => jest.clearAllMocks());
+
+    it('리액션 저장', async () => {
+      const user = {
+        id: 1,
+        email: 'test1',
+        nickname: 'test1',
+        profileImage: null,
+      } as User;
+      const diaryId = 1;
+      const diary = {
+        id: 1,
+        title: 'title',
+        content: '<p>content</p>',
+        summary: 'summary',
+        thumbnail: '',
+        emotion: '🥹',
+        mood: MoodDegree.SO_SO,
+        status: DiaryStatus.PUBLIC,
+      } as Diary;
+      const reactionRequestDto = { reaction: '💝' } as ReactionRequestDto;
+
+      (diariesService.findDiary as jest.Mock).mockResolvedValue(diary);
+      (reactionsRepository.findReactionByDiaryAndUser as jest.Mock).mockResolvedValue(null);
+
+      await reactionsService.saveReaction(user, diaryId, reactionRequestDto);
+      expect(reactionsRepository.save).toHaveBeenCalledTimes(1);
+      expect(reactionsRepository.addDiaryEvent).toHaveBeenCalledTimes(1);
+    });
+
+    it('이미 해당 일기에 리액션을 한 경우 예외 발생', () => {
+      const user = {
+        id: 1,
+        email: 'test1',
+        nickname: 'test1',
+        profileImage: null,
+      } as User;
+      const diaryId = 1;
+      const diary = {
+        id: 1,
+        title: 'title',
+        content: '<p>content</p>',
+        summary: 'summary',
+        thumbnail: '',
+        emotion: '🥹',
+        mood: MoodDegree.SO_SO,
+        status: DiaryStatus.PUBLIC,
+      } as Diary;
+      const reactionRequestDto = { reaction: '💝' } as ReactionRequestDto;
+      const reaction = { id: 1, reaction: '🔥' } as Reaction;
+
+      (diariesService.findDiary as jest.Mock).mockResolvedValue(diary);
+      (reactionsRepository.findReactionByDiaryAndUser as jest.Mock).mockResolvedValue(reaction);
+
+      expect(reactionsService.saveReaction(user, diaryId, reactionRequestDto)).rejects.toThrow(
+        new BadRequestException('이미 해당 글에 리액션을 남겼습니다.'),
+      );
+
+      expect(reactionsRepository.save).toHaveBeenCalledTimes(0);
+      expect(reactionsRepository.addDiaryEvent).toHaveBeenCalledTimes(0);
```

Comment:
```suggestion
      await expect(async() => await reactionsService.saveReaction(user, diaryId, reactionRequestDto)).rejects.toThrow(
        new BadRequestException('이미 해당 글에 리액션을 남겼습니다.'),
      );
      expect(reactionsRepository.save).toHaveBeenCalledTimes(0);
      expect(reactionsRepository.addDiaryEvent).toHaveBeenCalledTimes(0);
```
여기서 saveReaction에 await 처리를 하지 않으면 아마 그 아래에 있는 expect 횟수 검증이 제대로 동작하지 않을거에요.

Reply:
아 그렇겠군여..!! 다른 부분도 같이 수정해놓을게요.